compiler_opt/rl/corpus.py

```diff
@@ -34,77 +35,93 @@ def build_modulespecs_from_datapath(
     additional_flags: Tuple[str, ...] = (),
     delete_flags: Tuple[str, ...] = ()
 ) -> List[ModuleSpec]:
-  module_paths: List[str] = _load_module_paths(data_path)
-
-  has_thinlto: bool = _has_thinlto_index(module_paths)
+  # TODO: (b/233935329) Per-corpus *fdo profile paths can be read into
+  # {additional|delete}_flags here
+  metadata: Dict[str, any] = _load_metadata(
+      os.path.join(data_path, 'metadata.json'))
+  module_paths = metadata['modules']
+  if len(module_paths) == 0:
+    raise ValueError(f'{data_path}\'s metadata contains no modules.')
+
+  has_thinlto: bool = metadata['has_thinlto']
+
+  # Future: cmd_override could have per-module overrides if needed
```

Comment:
We don't have a scenario for that, do we?

Reply:
meant it as a note for "if you ever want per module overrides, here's where to do it", I can only think of if x flag exists, override with y set of options, which sounds plausible enough

Reply:
I'd remove the comment though - if we hit the scenario in the future, we'll figure it out (i.e. we'll see what exactly the scenario entails, and, surely enough, if it fits with what's here, great).